ipn/serve.go

```diff
@@ -44,6 +44,12 @@ type ServeConfig struct {
 	// of either the client or the LocalBackend does not expose ports
 	// that users are not aware of.
 	Foreground map[string]*ServeConfig `json:",omitempty"`
+
+	// Etag is the checksum of the serve config that's populated
+	// by the LocalClient through the HTTP Etag header during a
+	// GetServeConfig request and is translated to an If-Match header
+	// during a SetServeConfig request.
+	Etag string `json:"-"`
```

Comment:
I'd go with the convention ETag here. It's just turned into "Etag" on the wire for HTTP/1.1 and in Go maps.